Source/Core/VideoBackends/OGL/GLInterface/EGL.cpp

```diff
@@ -29,52 +29,68 @@ void cInterfaceEGL::DetectMode()
 	EGLint num_configs;
 	EGLConfig *config = nullptr;
 	bool supportsGL = false, supportsGLES2 = false, supportsGLES3 = false;
+	std::array<int, 4> renderable_types = {
+		EGL_OPENGL_BIT,
+		(1 << 6), /* EGL_OPENGL_ES3_BIT_KHR */
+		EGL_OPENGL_ES2_BIT,
+	};
 
-	// attributes for a visual in RGBA format with at least
-	// 8 bits per color
-	int attribs[] = {
-		EGL_RED_SIZE, 8,
-		EGL_GREEN_SIZE, 8,
-		EGL_BLUE_SIZE, 8,
-		EGL_NONE };
-
-	// Get how many configs there are
-	if (!eglChooseConfig( egl_dpy, attribs, nullptr, 0, &num_configs))
+	for (auto renderable_type : renderable_types)
 	{
-		INFO_LOG(VIDEO, "Error: couldn't get an EGL visual config\n");
-		goto err_exit;
-	}
+		// attributes for a visual in RGBA format with at least
+		// 8 bits per color
+		int attribs[] = {
+			EGL_RED_SIZE, 8,
+			EGL_GREEN_SIZE, 8,
+			EGL_BLUE_SIZE, 8,
+			EGL_RENDERABLE_TYPE, renderable_type,
```

Comment:
Problem is with that, is that they could potentially only return configurations that match all of the bits requested, which would also cause issues.